### PR TITLE
Docs: Fix missing options for remove_orphan_files procedure

### PR DESCRIPTION
**Pull request by manuzhang**
cc @karuppayya

### CODE REVIEW

File: docs/docs/spark-procedures.md [hunk 1]
```diff
@@ -312,6 +312,10 @@ Used to remove files which are not referenced in any metadata files of an Iceber
 | `location`    |    | string    | Directory to look for files in (defaults to the table's location) |
 | `dry_run`     |    | boolean   | When true, don't actually remove files (defaults to false) |
 | `max_concurrent_deletes` |    | int       | Size of the thread pool used for delete file actions (by default, no thread pool is used) |
+| `file_list_view` |    | string | Dataset to look for files in (skipping the directory listing) |
+| `equal_schemes` |    | map<string, string> | Mapping of file system schemes to be considered equal (default to `map('s3a,s3n','s3')`). Key can be a comma-separated list. |
```
Comment (karuppayya) [inline docs/docs/spark-procedures.md:316]: nit: default -> defaults

File: docs/docs/spark-procedures.md [hunk 2]
```diff
@@ -312,6 +312,10 @@ Used to remove files which are not referenced in any metadata files of an Iceber
...
+| `equal_authorities` |    | map<string, string> | Mapping of file system authorities to be considered equal. Key can be a comma-separated list. |
+| `prefix_mismatch_type` |    | string | Action behavior when location prefixes (schemes/authorities) mismatch: <ul><li>ERROR - throw an exception. (default) </li><li>IGNORE - no action.</li><li>DELETE - delete files.</li></ul> |
```
Comment (karuppayya) [inline docs/docs/spark-procedures.md:318]: prefix_mismatch_type -> prefix_mismatch_mode

File: docs/docs/spark-procedures.md [hunk 3]
```diff
@@ -331,6 +335,50 @@ Remove any files in the `tablelocation/data` folder which are not known to the t
...
+        .withColumnRenamed("lastModified", "last_modified");
+String fileListViewName = "files_view";
+compareToFileList.createOrReplaceTempView(fileListViewName);
+```
+```sql
+CALL catalog_name.system.remove_orphan_files(table => 'db.sample', file_list_view => 'files_view');
+```
+
+When a file matches references in metadata files except for location prefix (scheme/authority), an error is thrown by default. 
+We can ignore the error and skip the file by setting `prefix_mismatch_type` to `IGNORE`.
```
Comment (karuppayya) [inline docs/docs/spark-procedures.md:353]: prefix_mismatch_type -> prefix_mismatch_mode

File: docs/docs/spark-procedures.md [hunk 4]
```diff
@@ -312,6 +312,10 @@ Used to remove files which are not referenced in any metadata files of an Iceber
 | `location`    |    | string    | Directory to look for files in (defaults to the table's location) |
 | `dry_run`     |    | boolean   | When true, don't actually remove files (defaults to false) |
 | `max_concurrent_deletes` |    | int       | Size of the thread pool used for delete file actions (by default, no thread pool is used) |
+| `file_list_view` |    | string | Dataset to look for files in (skipping the directory listing) |
+| `equal_schemes` |    | map<string, string> | Mapping of file system schemes to be considered equal (defaults to `map('s3a,s3n','s3')`). Key can be a comma-separated list. |
```
Comment (karuppayya) [inline docs/docs/spark-procedures.md:316]: since it is a map, should we clarify the key and value is.

**Review by karuppayya — COMMENTED** [review]
left a minor comment, otherwise lgtm

**Comment by manuzhang** [comment]
@karuppayya @amogh-jahagirdar Any more comments?

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks @manuzhang this looks good now!

**Comment by manuzhang** [comment]
Thanks @amogh-jahagirdar, I submit another PR https://github.com/apache/iceberg/pull/11133 to back-port the fix to previous versions.

**Pull request merged**